plugins/plugin-chart-echarts/src/Timeseries/transformProps.ts

```diff
@@ -127,24 +144,140 @@ export default function transformProps(chartProps: ChartProps): EchartsTimeserie
       show: annotationShow,
       style,
     } = layer;
-    if (annotationShow && isFormulaAnnotationLayer(layer)) {
-      series.push({
-        name,
-        id: name,
-        itemStyle: {
-          color: color || colorFn(name),
-        },
-        lineStyle: {
-          opacity: parseAnnotationOpacity(annotationOpacity),
-          type: style,
-          width: annotationWidth,
-        },
-        type: 'line',
-        smooth: true,
-        // @ts-ignore
-        data: evalFormula(layer, data),
-        symbolSize: 0,
-      });
+    if (annotationShow) {
```

Comment:
These if branches (and the transform function itself) are growing a little bit too big, do you think if it makes sense to split some pieces into smaller functions? For example, maybe each annotation transformer can have its own function?

Reply:
You're right, this was probably overdue, will split it up.